allennlp/models/crf_tagger.py

```diff
@@ -81,20 +104,45 @@ def __init__(self, vocab: Vocabulary,
                                                            self.num_tags))
 
         if constraint_type is not None:
+            warnings.warn("'constraint_type' was removed and replaced with"
+                          "'label_encoding', 'constrain_crf_decoding', and "
+                          "'calculate_span_f1' in version 0.6.1. It will be "
+                          "removed in version 0.8.", DeprecationWarning)
```

Comment:
how do we track things we intend to remove?

Reply:
Unfortunately, it's a pre-release sort of thing 😢 . Perhaps we could start an issue for each thing to deprecate and tag it with the milestone of the release that it's meant to be removed in? Then, before we make that release, we go through and make sure that everything that needs to be removed has been removed?

Reply:
Yes, I think that's what we need to do.

Reply:
https://github.com/allenai/allennlp/milestone/7

Reply:
should we remove in 0.7 or 0.8? I opted for 0.8 since other big projects (e.g., [scikit-learn](http://scikit-learn.org/stable/developers/contributing.html#deprecation)) deprecate in 2 releases, and say we decide to deprecate something in the last 0.6.x release --- deprecating in 0.7 means that there's almost no warning.